CMakeLists.txt

```diff
@@ -34,7 +34,8 @@ endif()
 
 option(BUILD_SHARED_LIBS "Build shared libraries." ON)
 option(DLAF_WITH_OPENMP "${DLAF_WITH_OPENMP_DESCRIPTION}" ${DLAF_WITH_OPENMP_DEFAULT})
-option(DLAF_WITH_MKL "Enable MKL as provider for LAPACK" OFF)
+option(DLAF_WITH_MKL "Enable oneMKL as provider for LAPACK" OFF)
+option(DLAF_WITH_MKL_LEGACY "Enable MKL as provider for LAPACK" OFF)
```

Comment:
Not requesting a change, but I guess another option would be to have a single option, with three values: `OFF`, `LEGACY`, `ONEMKL`. I suppose it doesn't matter too much, both should be equally easy to model in spack. Just wanted to check if you considered this option and concluded it was worse?

Reply:
Another option is to have `DLAF_WITH_MKL_LEGACY` be a dependent option of `DLAF_WITH_MKL`. Again, not requesting a change, just presenting an alternative, in case you like it more.

Reply:
Thanks for the suggestion. It ended up like it is now without thinking too much actually.

The only pro to multi-value option I can see is that it make by construction mutual-exclusive the selection, for the rest I don't see pro-cons at the moment.

Let's see what @rasolca thinks about it 😉

Reply:
The idea is to keep the legacy just for a while and then delete it.
I think that using two variables can simplify the removal.